coverage/PENDING_SUBMIT/amdvlk/0564-COVERAGE-value-handle-h-470-505-114/cov-function-nested-loops-break-early-never-discard.amber

```diff
@@ -0,0 +1,435 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific LLVM code paths
+
+# The test passes because the shader always writes red.
+
+# Optimized using spirv-opt with the following arguments:
+# '--convert-local-access-chains'
+# '--copy-propagate-arrays'
+# '--reduce-load-size'
+# '--scalar-replacement=100'
+# '--eliminate-dead-code-aggressive'
+# '--eliminate-dead-inserts'
+# '--convert-local-access-chains'
+# '--private-to-local'
+# '--eliminate-local-single-block'
+# '--combine-access-chains'
+# '--eliminate-local-single-block'
+# '--eliminate-local-single-store'
+# '--eliminate-dead-branches'
+# '--merge-return'
+# '--eliminate-dead-branches'
+# '--eliminate-local-single-store'
+# '--ccp'
+# '--simplify-instructions'
+# '--eliminate-dead-inserts'
+# '--inline-entry-points-exhaustive'
+# '--simplify-instructions'
+# '--eliminate-dead-branches'
+# '--merge-blocks'
+# '--eliminate-dead-branches'
+# '--merge-return'
+# '--ccp'
+# spirv-opt commit hash: a0370efd589be33d5d9a85cfde2f85841b3755af
+
+
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 320 es
+# #define _int_20 _GLF_uniform_int_values[0]
+# #define _int_5 _GLF_uniform_int_values[1]
+# #define _int_0 _GLF_uniform_int_values[2]
+# #define _int_1 _GLF_uniform_int_values[3]
+# #define _float_1_0 _GLF_uniform_float_values[0]
+#
+# precision highp float;
+# precision highp int;
+#
+# // Contents of _GLF_uniform_float_values: 1.0
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     float _GLF_uniform_float_values[1];
+# };
+#
+# // Contents of _GLF_uniform_int_values: [20, 5, 0, 1]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     int _GLF_uniform_int_values[4];
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# int func()
+# {
+#     int a = _int_0;
+#
+#     for(int i = _int_1; i < _int_20; i++)
+#     {
+#         // Always false.
+#         if(gl_FragCoord.y < 0.0)
+#         {
+#             discard;
+#         }
+#
+#         for(int j = 1; j < 80; j ++)
```

Comment:
Is it possible to decrease these loop bounds? 20*80 inner loop iterations is rather high.

Reply:
The inner loop is only executed five times in total because of _a_.